progressbar.go

```diff
@@ -255,17 +274,13 @@ func (p *ProgressBar) Finish() error {
 	if p == nil {
 		return fmt.Errorf("progressbar is nil")
 	}
+
```

Comment:
So, we are giving option to library user to enable```OptionIgnoreLength()``` which will do following  
```golang
p.config.predictTime = false
p.config.showIterationsCount = false
p.config.ignoreLength = true
```
Some options could be enabled if called after this option from ```main```. It will depend on what sequence of options are given, which would not be a correct way. Example: 

```golang
bar := progressbar.NewOptions(
		-1,
		OptionSetTheme(Theme{Saucer: "*", SaucerPadding: " ", BarStart: "|", BarEnd: "|"}),
		OptionIgnoreLength(),
                 // predict time was enabled
		OptionSetPredictTime(true),
	)
```

I would propose that in ```NewOptions64()``` method, check if ```b.config.max==-1``` then disable required parameters and enable ignoreLength. 

```golang
if b.config.max==-1{
   // means we don't know the length 
   b.config.max = int64(p.config.width)
   b.config.predictTime = false
   b.config.showIterationsCount = false
   b.config.ignoreLength = true
}
```

Required flag enabled securely without providing option to user.